src/main/java/io/cdap/plugin/gcp/dataplex/common/config/DataplexBaseConfig.java

```diff
@@ -0,0 +1,244 @@
+   */
+  public boolean autoServiceAccountUnavailable() {
+    if (connection == null || connection.getServiceAccountFilePath() == null &&
+      connection.isServiceAccountFilePath()) {
+      try {
+        ServiceAccountCredentials.getApplicationDefault();
+      } catch (IOException e) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * @return GoogleCredentials based on service account
+   */
+  public GoogleCredentials getCredentials() {
+    GoogleCredentials credentials = null;
+    try {
+      credentials = getCredentialsFromServiceAccount();
+    } catch (IOException e) {
+      LOG.debug("Unable to load service account credentials due to error: {}", e.getMessage());
+    }
+    return credentials;
+  }
+
+  /**
+   * @return Project ID
+   */
+  public String getProject() {
+    if (connection == null) {
+      throw new IllegalArgumentException(
+        "Could not get project information, connection should not be null!");
+    }
+    return connection.getProject();
+  }
+
+  public GCPConnectorConfig getConnection() {
+    return connection;
+  }
+
+  @Nullable
+  public String tryGetProject() {
+    return connection == null ? null : connection.tryGetProject();
+  }
+
+  @Nullable
+  public String getServiceAccount() {
+    return connection == null ? null : connection.getServiceAccount();
+  }
+
+  @Nullable
+  public String getServiceAccountType() {
+    return connection == null ? null : connection.getServiceAccountType();
+  }
+
+  @Nullable
+  public Boolean isServiceAccountFilePath() {
+    return connection == null ? null : connection.isServiceAccountFilePath();
+  }
+
+  /**
+   * Check if service account is a valid one.
+   *
+   * @param failureCollector
+   */
+  public void validateServiceAccount(FailureCollector failureCollector) {
+    if (connection.isServiceAccountJson() || connection.getServiceAccountFilePath() != null) {
+      GoogleCredentials credentials = getCredentials();
+      if (credentials == null) {
+        failureCollector.addFailure(String.format("Unable to load credentials from %s.",
+              connection.isServiceAccountFilePath() ? connection.getServiceAccountFilePath() : "provided JSON key"),
+            "Ensure the service account file is available on the local filesystem.")
+          .withConfigProperty("serviceFilePath")
+          .withConfigProperty("serviceAccountJSON");
+        throw failureCollector.getOrThrowException();
+      }
+    }
+  }
+
+  private GoogleCredentials getCredentialsFromServiceAccount() throws IOException {
+    GoogleCredentials credentials = null;
+    // validate service account
+    if (connection.isServiceAccountJson() || connection.getServiceAccountFilePath() != null) {
+      credentials =
+        GCPUtils.loadServiceAccountCredentials(getServiceAccount(), isServiceAccountFilePath())
+          .createScoped(Lists.newArrayList(DataplexConstants.DATAPLEX_SCOPE));
```

Comment:
@yaojiefeng Since, the scope required by dataplex api is `cloud-platform` so `auto-detect` should work on ephemeral dataproc cluster. Since, we set the `cloud-platform` scope when creating it. https://github.com/cdapio/cdap/blob/c15d12f1876ef81b542ad5a4220dcae59ae3eb8a/cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java#L422
Here, when we set the scope we are just requesting the access token with those scopes while creating client but if the Compute Engine VM does not have those scopes set the request using the access token will fail.

Reply:
I see. Then there is no problem, then I don't think you need to add this, this will just be same as other gcp plugins

Reply:
It was failing while we were using REST api calls instead of client libraries but now it is working fine without this also. So removing it.

Reply:
I don't see this getting removed, is this still required now?

Reply:
I have removed it in latest commit.